activerecord/lib/active_record/associations/builder/has_and_belongs_to_many.rb

```diff
@@ -67,7 +72,17 @@ def middle_reflection(join_model)
                                 middle_options)
     end
 
+    private_class_method :valid_options, :validate_options
+
+    def self.validate_options(options)
+      super
+    end
```

Comment:
Do you need to define this method if all it is doing is calling `super`?

Reply:
Yeah, because the method in the base class is private, so I can't call it directly. The alternative would be to `send`, but I thought this was the least worst of the two options, but I'm happy to change to `self.class.send :validate_options, options`, or throw in a comment to explain why this is needed, lmk.

Reply:
Ack, y'know, turns out I fooled myself into thinking this was working, when actually I had the `private_class_method` call BEFORE I defined this method, making this a public class method :)  Let me change this to the `send` option.